### PR TITLE
fix findbugs warning 'No relationship between generic parameter and m…

### PR DESCRIPTION
**Pull request by jhrcek**
fix findbugs warning 'No relationship between generic parameter and method argument'
fixed 4 instances of the warning. At 4 places method argument was incorrect (e.g. String where OrganizationalEntity was in the collection). While I was at it i enabled 1 test, reintroduced some asserts (that were likely disabled because of failing tests), and removed some boilerplate from tests. All affected tests are passing.

### CODE REVIEW

File: jbpm-flow/src/main/java/org/jbpm/workflow/core/node/StateNode.java [hunk 1]
```diff
@@ -52,8 +52,8 @@ public void addConstraint(ConnectionRef connectionRef, Constraint constraint) {
         constraints.put(connectionRef, constraint);
     }
     
-    public Constraint getConstraint(String name){
-        return constraints.get(name);
+    public Constraint getConstraint(ConnectionRef connectionRef){
+        return constraints.get(connectionRef);
```
Comment (jhrcek) [inline jbpm-flow/src/main/java/org/jbpm/workflow/core/node/StateNode.java]: constraints is Map<ConnectionRef, Constraint>

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/LifeCycleBaseTest.java [hunk 2]
```diff
@@ -1245,10 +1240,9 @@ public void testForwardFromReady() throws Exception {
 
         taskService.forward(taskId, "Darth Vader", "Tony Stark");
 
-
         Task task2 = taskService.getTaskById(taskId);
-        assertFalse(task2.getPeopleAssignments().getPotentialOwners().contains("Darth Vader"));
```
Comment (jhrcek) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/LifeCycleBaseTest.java]: The main reason for this refactoring getPotentialOwners returns List<OrganizationalEntity>, so cannot contain String. Also enabled this method as @Test

File: jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/util/PeopleAssignmentHelperTest.java [hunk 3]
```diff
@@ -55,39 +55,37 @@ public void testProcessPeopleAssignments() {
 		List<OrganizationalEntity> organizationalEntities = new ArrayList<OrganizationalEntity>();
 		
 		String ids = "espiegelberg,   drbug   ";
-		assertTrue(organizationalEntities.size() == 0);		
+		assertTrue(organizationalEntities.isEmpty());
 		peopleAssignmentHelper.processPeopleAssignments(ids, organizationalEntities, true);
 		assertTrue(organizationalEntities.size() == 2);
-		organizationalEntities.contains("drbug");
```
Comment (jhrcek) [inline jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/util/PeopleAssignmentHelperTest.java]: organizationalEntities is List<OrganizationalEntities> so cannot contain String.

File: jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/util/PeopleAssignmentHelperTest.java [hunk 4]
```diff
@@ -55,39 +55,37 @@ public void testProcessPeopleAssignments() {
...
-		assertTrue(organizationalEntities.get(0) instanceof Group);
+		assertTrue(organizationalEntities.contains(createGroup("Software Developer")));
 		
 		// Test that a duplicate is not added; only 1 of the 2 passed in should be added
 		ids = "Software Developer,Project Manager";
 		peopleAssignmentHelper.processPeopleAssignments(ids, organizationalEntities, false);
 		assertTrue(organizationalEntities.size() == 2);
-		assertTrue(organizationalEntities.get(0) instanceof Group);
-		assertTrue(organizationalEntities.get(1) instanceof Group);
+		assertTrue(organizationalEntities.contains(createGroup("Software Developer")));
```
Comment (jhrcek) [inline jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/util/PeopleAssignmentHelperTest.java]: Stronger asserts

File: jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/timer/GlobalJPATimerJobFactoryManager.java [hunk 5]
```diff
@@ -120,7 +120,7 @@ public void removeTimerJobInstance(TimerJobInstance instance) {
         return singleTimerInstances.values();
     }
     
-    public Collection<TimerJobInstance> getTimerJobInstances(Integer sessionId) {
+    public Collection<TimerJobInstance> getTimerJobInstances(Long sessionId) {
```
Comment (jhrcek) [inline jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/timer/GlobalJPATimerJobFactoryManager.java]: timerInstances is Map<Long, Map<Long, TimerJobInstance>>

**Comment by mswiderski** [comment]
merged into master, thanks @jhrcek

**Pull request closed without merging**